Marlin/src/gcode/motion/G2_G3.cpp

```diff
@@ -125,7 +125,9 @@ void plan_arc(
   // Vector rotation matrix values
   float raw[XYZE];
   const float theta_per_segment = angular_travel / segments,
-              linear_per_segment = linear_travel / segments,
+              #if DISABLED(AUTO_BED_LEVELING_UBL)
+                linear_per_segment = linear_travel / segments,
```

Comment:
This ought to be getting revived soon. Arcs are supposed to have a linear axis that can move as the arc is drawn. For example, you should be able to make a circle in the XY plane and have a Z move at the same time. The change that removed this functionality for UBL is a temporary hack as a workaround for UBL not being fully integrated with the planner. I will study the problem and try to provide a proper fix soon.

Reply:
That's ok. such function you talk about is often ised in CNC to make threads